core/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobContainer.java

```diff
@@ -95,14 +97,17 @@ public boolean blobExists(String blobName) {
 
     @Override
     public InputStream readBlob(String name) throws IOException {
-        return new BufferedInputStream(Files.newInputStream(path.resolve(name)), blobStore.bufferSizeInBytes());
+        final Path resolvedPath = path.resolve(name);
+        if (!Files.exists(resolvedPath)) {
```

Comment:
I am not sure I understand why this check is necessary. Could you add a comment that clarifies it?